gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,229 @@
+            func: "{that}.getInstalledSolutionsForCurrentDevice"
+        },
+        "onCaptureSettingsForCurrentDevice.captureSystemSettings": {
+            func: "{that}.captureSystemSettings",
+            args: ["{arguments}.0", "{arguments}.1"],
+            priority: "after:getInstalledSolutionsForCurrentDevice"
+        },
+        "onCaptureSettingsForCurrentDevice.formatRawCapturedSettings": {
+            func: "gpii.flowManager.formatRawCapturedSettings",
+            args: ["{arguments}.0"],
+            priority: "after:captureSystemSettings"
+        }
+    },
+    invokers: {
+        getInstalledSolutionsForCurrentDevice: {
+            funcName: "gpii.flowManager.getInstalledSolutionsForCurrentDevice",
+            args: ["{that}"]
+        },
+        getSystemSettingsCapture: {
+            funcName: "gpii.flowManager.getSystemSettingsCapture",
+            args: ["{that}.events.onCaptureSettingsForCurrentDevice", "{arguments}.0"] // options
+        },
+        captureSystemSettings: {
+            funcName: "gpii.flowManager.captureSystemSettings",
+            args: ["{lifecycleManager}.read", "{arguments}.0", "{arguments}.1"] // solutionsRegistryEntries, options
+        }
+    }
+});
+
+gpii.flowManager.getInstalledSolutionsForCurrentDevice = function (that) {
+    return fluid.promise.fireTransformEvent(that.events.onSolutionsForCurrentDevice);
+};
+
+/**
+ * This main API entry point for capturing settings from a system or computer. This captures
+ * the actual settings on the device, so it assumed to be running in a local untrusted flow
+ * manager.
+ *
+ * @param {Event} onCaptureSettingsForCurrentDevice - The transforming promise chain
+ * @param {Object} options - Options for this chain.
+ * @param {Array} options.solutionsList - An array of solution IDs to filter by when
+ * retreiving settings. If this option is not included, all available settings will be
+ * returned. ex: `["com.microsoft.windows.mouseSettings", "com.freedomscientific.jaws"]`.
+ * @return {Promise} A promise resolved with the payload of captured system settings.
+ */
+gpii.flowManager.getSystemSettingsCapture = function (onCaptureSettingsForCurrentDevice, options) {
+    return fluid.promise.fireTransformEvent(onCaptureSettingsForCurrentDevice,
+        {}, options);
+};
+
+/**
+ * This function invokers a settings handler to fetch settings, but wraps the processing and reports
+ * back any failures in an `isError: true` json block.
+ *
+ * @param {Function} invokeSettingsHandlerGet - The get function from the appropriate settings handler.
+ * @param {String} solutionID - The dotted solution id, ex `com.freedomscientific.jaws`
+ * @param {Object} handlerSpec - TODO, exactly how much of the solutions entry block is this?
+ * @return {Object} The returned settings, or an error block with debugging information.
+ */
+gpii.flowManager.capture.safeHandlerGet = function (invokeSettingsHandlerGet, solutionID, handlerSpec) {
```

Comment:
This function seems no longer in use. Please double check. Thanks.

Reply:
Good catch, removed this.